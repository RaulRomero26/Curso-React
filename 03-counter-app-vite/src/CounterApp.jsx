import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) =>{
    //se le pasas como paramtro el de las props, para usarlo, su valor inicial es el que mandamos desde el padre

    const [ counter, setCounter ] = useState( value ) //use state crea la propiedad en las dev tool, el primer valor desestructurado es el nombre que  le queremos poner a ese valor, el segundo valor es el nombre de la funcion mediante la cual queremos mutar el valor
                                                  // notese que en ningun momento se "definio" la funcion solo se usa y hace lo de dentro de los parentesis le dice a react, el estado cambio, redibuja el componente

    const handleAdd = () => {
        //console.log(event)
        setCounter(counter+1)
        //setCounter( (c) => c+1 ) es el mismo
    }

    const handleLess = () => {
        setCounter(counter-1)
    }

    const handleReset = () => {
        setCounter(value)//el mismo de las props
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleLess }>-1</button>
            <button aria-label="btn-reset" onClick={ handleReset }>Reset</button>
        </>
    );

}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0,
}