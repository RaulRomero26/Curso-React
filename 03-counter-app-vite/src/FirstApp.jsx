import PropTypes from 'prop-types';
/*
const newMessage = {
    message: 'Hola Mundo',
    title: 'RaÃºl'
}

const getMessage = (message) => {
    return message.title;
}*/

export const FirstApp = ({ title, subtitle, name }) =>{

    // if(!title){
    //     throw new Error('No existe el titulo')
    // }
    // //console.log(props);

    return(
        <>
            <h1 data-testid="test-title">{ title }</h1>
            {/* <code>{ JSON.stringify(newMessage) } </code> */}
            <p>{ subtitle }</p>
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    name: 'Raul Romero',
    subtitle: 'No hay subtitulo'
    // title: 'No hay titulo',
}