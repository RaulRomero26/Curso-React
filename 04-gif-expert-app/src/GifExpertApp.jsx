import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
// import React from 'react'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Programacion','Dragon Ball'])

    const onAddCategory = ( onNewCategory ) =>{
        console.log(onNewCategory);
        setCategories([ onNewCategory, ...categories ]);
        //setCategories(cat => [...cat,'Valorant'] );
    }

  return (
    <>
        <h1>GifExpertApp</h1>

        <AddCategory 
            //setCategories={ setCategories}{/*PROPIEDAD QUE HACE REFERENCIA A UNA FUNCION del padre se pasa al hijo*/}
            onNewCategory = { event => onAddCategory(event) } //esto es una propiedad
        /> 
        <ol>
            {
                categories.map(category => {
                    return <li key={category}>{category}</li>
                })
            }
           
        </ol>
    </>
  )
}
