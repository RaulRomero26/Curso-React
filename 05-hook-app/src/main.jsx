import React from 'react';
import ReactDOM from 'react-dom/client';
import { MemoHook } from './06-memos/MemoHook';
//import Memorize from './06-memos/Memorize';
//import { Layout } from './05-useLayoutEffect/Layout';
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
//import { CounterApp } from './01-useState/CounterApp';
//import { HooksApp } from './HooksApp';
//import { FocusScreen } from './04-useRef/FocusScreen';


import './index.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  //<React.StrictMode>
    <MemoHook />
  //</React.StrictMode>,
)
