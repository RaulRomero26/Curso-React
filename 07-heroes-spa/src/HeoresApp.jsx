

const HeoresApp = () => {
  return (
    <>
        <h1>heroees app</h1>
    </>
  )
}

export default HeoresApp
