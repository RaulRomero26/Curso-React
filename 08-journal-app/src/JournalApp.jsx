

export const JournalApp = () => {
  return (
    <>
        <h2>JournalApp</h2>
    </>
  )
}
