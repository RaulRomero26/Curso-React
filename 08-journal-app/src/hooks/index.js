export * from './useForm';
export * from './useChekAuth';