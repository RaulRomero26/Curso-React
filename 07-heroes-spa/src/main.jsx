import React from 'react';
import ReactDOM from 'react-dom/client';
import HeoresApp from './HeoresApp';
import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <HeoresApp />
  </React.StrictMode>,
);
