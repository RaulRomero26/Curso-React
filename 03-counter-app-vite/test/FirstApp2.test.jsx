import { render,screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en <FirsApp />', () => {

    const title = 'hola, soy goku'
    const subtitle = 'soy un subtitulo'
    
   test('debe de hacer match con el snapshot', () => {

        const { container } = render( <FirstApp title={title}/>)
        expect(container).toMatchSnapshot();
   });

   test('debe de mostrar el mensaje "hola, soy goku', () =>{

        render( <FirstApp title={title}/>)
        expect( screen.getByText(title)).toBeTruthy();
        //screen.debug() sirve para ver el snapshot
   }); 

   test('debe de mostrar el titulo en un h1', () => {

        render( <FirstApp title={title}/>)
        expect(screen.getByRole('heading',{level: 1}).innerHTML).toBe(title); // o toContain
   });

   test('debe de mostrar el subtitulo enviado por props',() => {

       render( <FirstApp 
                   title={title}
                   subtitle={subtitle}
               />)
       expect(screen.getAllByText(subtitle).length).toBe(2);
   })

});