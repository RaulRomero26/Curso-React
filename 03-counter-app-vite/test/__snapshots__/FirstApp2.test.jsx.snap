// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirsApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola, soy goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Raul Romero
  </p>
</div>
`;
